src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp

```diff
@@ -887,8 +888,32 @@ namespace
                 Spectrum ray_transmission;
                 phase_function->evaluate_transmission(
                     volume_ray, vertex.m_phase_function_data, ray_transmission);
+                const size_t channel_count = ray_transmission.size();
+                float density = 0.0f;
+                for (int i = 0; i < channel_count; ++i)
+                {
+                    if (density <= 1.0f - ray_transmission[i])
+                        density = 1.0f - ray_transmission[i];
+                }
+
+                const Spectrum& scattering_coef = phase_function->scattering_coefficient(
+                    volume_ray, vertex.m_phase_function_data);
+
+                const Spectrum& extinction_coef = phase_function->extinction_coefficient(
+                    volume_ray, vertex.m_phase_function_data);
 
-                const float density = 1.0f - min_value(ray_transmission);
+                // Precompute MIS weights.
+                Spectrum mis_weights(1.0f);
```

Comment:
I suggest initializing `mis_weights` with `scattering_coef`` to avoid a bunch of multiplications.